### PR TITLE
feat(streaming): integrate append-only executors to stream graph

### LINKED ISSUE #1550: Append-only HashAgg Executor
Implement append only hash aggregation, think about reuse code in batch engine.

### LINKED ISSUE #2407: Append-only HashJoin Executor

### PR DESCRIPTION
**Pull request by StrikeW**
## What's changed and what's your intention?

Integrate append-only version executors (TopN, HashAgg, HashJoin, SimpleAgg) to the stream graph.

## Checklist

- [x] I have written necessary docs and comments
- [x] I have added necessary unit tests and integration tests
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)

## Refer to a related PR or issue link (optional)
close #2407
close #1550

### CODE REVIEW

File: src/frontend/src/optimizer/plan_node/stream_topn.rs [hunk 1]
```diff
@@ -49,13 +49,21 @@ impl StreamTopN {
...
-            "StreamTopN {{ order: {}, limit: {}, offset: {} }}",
-            self.logical.topn_order(),
-            self.logical.limit(),
-            self.logical.offset(),
-        )
+        let mut builder = if self.input().append_only() {
+            f.debug_struct("StreamAppendOnlyTopN")
+        } else {
+            f.debug_struct("StreamTopN")
+        };
```
Comment (fuyufjh) [inline src/frontend/src/optimizer/plan_node/stream_topn.rs:56]: It looks duplicated with "append_only = true/false" in L62~64

(Similar for other operators)

Comment (StrikeW) [reply src/frontend/src/optimizer/plan_node/stream_topn.rs:56]: The `append_only ` attribute in `PlanBase` indicates whether the output of operator is append-only, it doesn't mean the operator is append-only optimized. So if we want to check whether the append-only version operator is used, we should differentiate it by name (likes the `StreamDeltaJoin` and `StreamHashJoin`).

Comment (fuyufjh) [reply src/frontend/src/optimizer/plan_node/stream_topn.rs:56]: Got it. In this way, I would recommend to remove  "append_only = true/false" in explain results.

**Review by fuyufjh — APPROVED** [review]

File: src/frontend/src/optimizer/plan_node/stream_topn.rs (hunk 1 above)
Comment (StrikeW) [reply src/frontend/src/optimizer/plan_node/stream_topn.rs:56]: agreed. I got it wrong before.

**Comment by skyzh** [comment]
Still no e2e tests? 🤣 Maybe add next time.

**Pull request merged**